### DISCUSSION TITLE
How do use the stage config file while using the powershell module

### DISCUSSION THREAD

**Discussion by nikhil-neu** [Azure-Player/azure.datafactory.tools#182] [General]
How do i use the stage the config file while using the powershell file 

I originally was using the ADO tasks , like this 
`                - task: PublishADFTask@1
                  enabled: true
                  inputs:
                    azureSubscription: 'AMN_Shared_Dev_Staffing_dev'
                    ResourceGroupName: '$(SharedRGName)'
                    DataFactoryName: '$(DataFactoryName)'
                    DataFactoryCodePath: '.'
                    Location: '$(location)'
                    StageType: 'FilePath'
                    StageConfigFile: '$(System.DefaultWorkingDirectory)/deployment/adf-config.json'
                    StopStartTriggers: false
                    FilteringType: 'Inline'
                    FilterText: '$(ADFOptions)'
                    PublishMethod: 'AzResource'`

Now i wanted change to equivalent powershell module for Publish , since i a moving to an ubuntu based self hosted build agent .

I dont see any option to pass the config file as the parameter

**Comment by NowinskiK** [comment]
The parameter you are looking for is `Stage`.
> The parameter can be either full path to csv file (must ends with .csv) or just stage name. 
Doc: https://github.com/SQLPlayer/azure.datafactory.tools#stage-parameter

**Reply by nikhil-neu** [reply]
@NowinskiK Thanks for that. I am having another issues , in the json config file , i have parameterized some of the values using the environment variable notation , like this ..

`"akv_ls_amnoneshd": [
      {
        "name": "$.properties.typeProperties.baseUrl",
        "value": "$($Env:kvURI)",
        "action": "update"
      }
    ],`
    
    This does not seem to work in ubuntu based agents , is there any work around for this ?
    I get the following error in ubuntu agents 


`Config file:   /home/vsts/work/1/s/deployment/adf-config.json
Update-PropertiesForObject: /home/vsts/.local/share/powershell/Modules/azure.datafactory.tools/0.95.1/private/Update-PropertiesFromFile.ps1:75
Line |
  75 |  …     $null = Update-PropertiesForObject -o $_ -action $action -path $p …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot bind argument to parameter 'value' because it is an
     | empty string.
`
